s3/s3.go

```diff
@@ -95,3 +95,29 @@ func (c *Client) PutObject(bucket, key string, body io.ReadSeeker) error {
 
 	return nil
 }
+
+func (c *Client) PutObjectWithMetadata(bucket, key string, body io.ReadSeeker, metadata map[string]*string) error {
```

Comment:
maybe we should have only one method PutObject and put metadata parameter as an optional, instead of having two separate method. I have the same issue on dynamodb with PutItem and trying to refactor it. Let's discuss about it.

Reply:
you're right, please take a look again